src/qibo/models/circuit.py

```diff
@@ -172,6 +181,23 @@ def __init__(self, nqubits, accelerators=None, density_matrix=False):
                 )
             self._distributed_init(nqubits, accelerators)
 
+        initial_queue = []
+
+        if eigenstate == "-":
+            initial_queue.append(gates.X)
+        elif eigenstate != "+":
+            raise NotImplementedError("Invalid eigenstate")
+        if initialize == "X":
+            initial_queue.append(gates.H)
+        elif initialize == "Y":
+            initial_queue.append(gates.H)
+            initial_queue.append(gates.S)
+        elif initialize != "Z":
+            raise NotImplementedError("Invalid initialize")
```

Comment:
I think that it is better to have a few comments to explain exactly what you are doing here. You could even create a separate method
```py
def _basis_initialization(self, basis/initialize : str, eigenstate : str):
```